src/math/math_base.h

```diff
@@ -105,5 +106,73 @@ struct PointBase3D : public PointBase2D<_Type>
     _Type z;
 };
 
+template <typename _Type>
+struct LineBase2D
+{
+    LineBase2D(const PointBase2D<_Type> & _p1, const PointBase2D<_Type> & _p2)
+        : p1( _p1 ), p2( _p2 )
+    { }
+
+    LineBase2D(const PointBase2D<_Type> & _p, double angleRadians)
+    {
+        this->p1 = PointBase2D<_Type>(_p.x, _p.y);
+        this->p2 = PointBase2D<_Type>(std::cos(angleRadians) * length(), std::sin(angleRadians) * length());
+    }
+
+    bool operator == ( const LineBase2D & line ) const
+    {
+        return p1 == line.p1 && p2 == line.p2;
+    }
+
+    LineBase2D & operator += ( const PointBase2D<_Type> & point )
+    {
+        return LineBase2D(this->p1 + point, this->p2 + point);
+    }
+
+    double length() const
+    {
+        const double x = static_cast<double>(p2.x) - static_cast<double>(p1.x);
+        const double y = static_cast<double>(p2.y) - static_cast<double>(p1.y);
+        return std::sqrt(x * x + y * y);
+    }
+
+    double angle() const
+    {
+        if (p1 == p2) throw std::invalid_argument("Points p1 and p2 must be different!");
```

Comment:
Let's add a function which validates line:
```cpp
bool valid() const
{
    return (p1 != p2);
}
```

Reply:
> Let's add a function which validates line:
> 
> ```c++
> bool valid() const
> {
>     return (p1 != p2);
> }
> ```

Not needed anymore as `atan2` now handles the zero case.